brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java

```diff
@@ -206,156 +190,8 @@ private void processBlockTransactions(Scheduler scheduler) {
-    final Map<String, Set<String>> accountRemovedSignatories = new HashMap<>();
-
-    commands
-        .stream()
-        .filter(Command::hasRemoveSignatory)
-        .map(Command::getRemoveSignatory)
-        .filter(command -> registeredAccounts.contains(command.getAccountId()))
-        .forEach(removeSignatory -> {
-          final String signatoryAccountId = removeSignatory.getAccountId();
-          if (!accountRemovedSignatories.containsKey(signatoryAccountId)) {
-            accountRemovedSignatories.put(signatoryAccountId, new HashSet<>());
-          }
-          accountRemovedSignatories.get(signatoryAccountId)
-              .add(removeSignatory.getPublicKey().toUpperCase());
-        });
-    return accountRemovedSignatories;
-  }
-
-  private Map<String, Set<String>> constructAddedSignatoriesByAccountId(
-      Collection<Command> commands,
-      Set<String> registeredAccounts) {
-    final Map<String, Set<String>> accountAddedSignatories = new HashMap<>();
-
-    commands
-        .stream()
-        .filter(Command::hasAddSignatory)
-        .map(Command::getAddSignatory)
-        .filter(command -> registeredAccounts.contains(command.getAccountId()))
-        .forEach(addSignatory -> {
-          final String signatoryAccountId = addSignatory.getAccountId();
-          if (!accountAddedSignatories.containsKey(signatoryAccountId)) {
-            accountAddedSignatories.put(signatoryAccountId, new HashSet<>());
-          }
-          accountAddedSignatories.get(signatoryAccountId)
-              .add(addSignatory.getPublicKey().toUpperCase());
-        });
-    return accountAddedSignatories;
-  }
-
-  private void registerCreatedAccountByTransactionScanning(Transaction blockTransaction)
-      throws InterruptedException {
-    List<String> createAccountList = blockTransaction
-        .getPayload()
-        .getReducedPayload()
-        .getCommandsList()
-        .stream()
-        .filter(Command::hasCreateAccount)
-        .map(Command::getCreateAccount)
-        .filter(command -> userDomains.contains(command.getDomainId()))
-        .map(command -> command.getAccountName().concat(accountIdDelimiter)
-            .concat(command.getDomainId()))
-        .collect(Collectors.toList());
-    if (!createAccountList.isEmpty()) {
-      final Set<String> userAccounts = registrationProvider.getUserAccounts();
-      createAccountList = createAccountList
-          .stream()
-          .filter(userAccounts::contains)
-          .collect(Collectors.toList());
-      registrationProvider.register(createAccountList);
-    }
-  }
-
-  private String getDomain(String accountId) {
-    try {
-      return accountId.split("@")[1];
-    } catch (Exception e) {
-      logger.warn("Couldn't parse domain of " + accountId, e);
-      return "";
+    if (blockTransactions != null && !blockTransactions.isEmpty()) {
+      pluggableLogicList.forEach(pluggableLogic -> pluggableLogic.apply(blockTransactions));
```

Comment:
What will happen if processing of transaction will throw an exception in one of pluggable Logics?

Reply:
The service is expected to die, since its state becomes invalid